### DISCUSSION TITLE
Pls help with select multiple

### DISCUSSION THREAD

**Discussion by tm-blg** [livewire/flux#666] [Feature Requests] [answered]
Hi, there! I'm trying to save array into model 

```
<flux:select wire:model.live="var.product_kinds" variant="listbox" multiple placeholder="Choose industries...">
        <flux:option>bulb</flux:option>
        <flux:option>partnumber</flux:option>
        <flux:option>projector</flux:option>
</flux:select>
```

```
#[Validate([
    'var.product_kinds' => 'array|nullable',
])]
public Feature $var;

public function mount()
{
    $this->var = new Feature();
}
```

Where i choose more than two option, i get LogicException "Something went wrong"

![image](https://github.com/user-attachments/assets/44212ac6-5b60-4079-8b2d-3682153aeac9)
![image](https://github.com/user-attachments/assets/f23959b8-7cda-4c7f-9e1b-0cf7b6d547c7)

**Comment by lotje-kinable** [comment]
It might have something to do with [legacy model binding](https://livewire.laravel.com/docs/upgrading#eloquent-model-binding) in Livewire 3. Model binding is replaced by Form objects.

**Comment by tm-blg** [comment]
Good Morning! legacy model binding helps me define $var
```

public Feature $var;

public function mount()
{
    $this->var = new Feature();
}
```

any changes in the model work well, except for the select...
I think I'm missing something and the answer is simple.

**Reply by jeffchown** [reply]
@tm-blg Did you make the update to the `config/livewire.php` file? (see below):

> In Livewire 3, binding directly to Eloquent models has been disabled in favor of using individual properties, or extracting [Form Objects](https://livewire.laravel.com/docs/forms#extracting-a-form-object).

> However, because this behavior is so heavily relied upon in Livewire applications, version 3 maintains support for this behavior via a configuration item in `config/livewire.php`:

`'legacy_model_binding' => true,`

> By setting legacy_model_binding to true, Livewire will handle Eloquent model properties exactly as it did in version 2.

**Reply by jeffchown** [reply]
Also, is the `product_kinds` column in your table a `string` or `text` cast as a `json`?

**Reply by tm-blg** [reply]
Thank you for ur answer!
i have `'legacy_model_binding' => true` I read about the difference in model binding in the second and third versions.

About product_kinds, I think updating the database and casts has nothing to do with it. I didn't even get to saving to the database.

In general, yes, I configured the casts in various ways.

tell me, can you reproduce my code in yours? For the purity of the experiment, the main thing is to save the data in the class.

**Reply by jeffchown** [reply]
@tm-blg I can't reproduce your issue without writing a lot of code to do so.
If you're able to post a full code sample, that recreates the issue, that can be copy/pasted, then it's easier to try to help.

**Reply by jeffchown** [reply]
@tm-blg Looking at the screenshot of the error again... Did you run `php artisan config:clear` after updating the config file?

**Reply by tm-blg** [reply]
For view:
```
<flux:select wire:model.live="var.product_kinds" variant="listbox" multiple placeholder="Choose industries...">
        <flux:option>bulb</flux:option>
        <flux:option>partnumber</flux:option>
        <flux:option>projector</flux:option>
</flux:select>
{{ var_dump($var->product_kinds)
```

For LiveWire:
```
#[Validate([
    'var.product_kinds' => 'array|nullable',
])]
public Feature $var;

public function mount()
{
    $this->var = new Feature();
}
```

Also you can change Model class or param name on your.

I would be extremely grateful to you!

**Reply by tm-blg** [reply]
> @tm-blgПосмотрите на скриншот ошибки еще раз... Вы запускали его `php artisan config:clear`после обновления файла конфигурации?

Ofcourse, php artisan optimize:clear

**Comment by joshhanley** [comment]
Closing as this isn't a feature request. If you have issues, please open an issue.

**Comment by tm-blg** [comment, marked as answer]
Hi, everyone! 

finally I found a solution.

This exception occurs either because of incorrectly specified casts.
In my case specifically it was incorrect validation. U just need add validation rule for everyone checkbox like this:

```
    public function rules()
    {
        return [
            **'test.one.*' => 'string|nullable',**
            'test.one' => 'array|nullable',
        ];
    }
```
